tensorizer/stream_io.py

```diff
@@ -1364,30 +1399,48 @@ def open_stream(
             # Not required to have been found,
             # and doesn't overwrite an explicitly specified endpoint.
             s3_endpoint = s3_endpoint or s3.s3_endpoint
-        except (ValueError, FileNotFoundError) as e:
-            # Uploads always require credentials here, but downloads may not
-            if is_s3_upload:
-                raise
-            else:
-                # Credentials may be absent because a public read
-                # bucket is being used, so try blank credentials,
-                # but provide a descriptive warning for future errors
-                # that may occur due to this exception being suppressed.
-                # Don't save the whole exception object since it holds
-                # a stack trace, which can interfere with garbage collection.
-                error_context = (
-                    "Warning: empty credentials were used for S3."
-                    f"\nReason: {e}"
-                    "\nIf the connection failed due to missing permissions"
-                    " (e.g. HTTP error 403), try providing credentials"
-                    " directly with the tensorizer.stream_io.open_stream()"
-                    " function."
-                )
-                s3_access_key_id = s3_access_key_id or ""
-                s3_secret_access_key = s3_access_key_id or ""
+
+            if force_http is None and s3.use_https is not None:
```

Comment:
This should follow after endpoint resolution as boto3 uses the `https://` or `http://` prefix on the configured endpoint respectively, which this logic should respect the same as the s3cmd's config version (which uses an explicit variable).

Essentially: https://github.com/boto/botocore/blob/a82eaae34b294970578e3af18d31242ac719ded0/botocore/session.py#L887C30-L892C36

Reply:
Will fix this. Planning this logic:
- If `force_http` is specified → use HTTP, otherwise
- If `.s3cfg` has `use_https = False` → use HTTP, otherwise
- If the endpoint is resolved by `botocore` or `AWS_ENDPOINT_URL` environment variables → use the endpoint's scheme, otherwise
- Use HTTPS

Reply:
This seems fine.

Reply:
Implemented in 8dfb67d88a14c9e8a90cce131b0702141514e2e5.